x-pack/plugins/ml/public/application/services/ml_api_service/index.ts

```diff
@@ -397,25 +397,45 @@ export function mlApiProvider(httpService: HttpService) {
       end,
       overallScore,
     }: {
-      jobId: string;
+      jobId: string[];
       topN: string;
       bucketSpan: string;
       start: number;
       end: number;
       overallScore?: number;
-    }) {
+    }): Promise<estypes.MlGetOverallBucketsResponse> {
       const body = JSON.stringify({
         topN,
         bucketSpan,
         start,
         end,
         ...(overallScore ? { overall_score: overallScore } : {}),
       });
-      return httpService.http<any>({
-        path: `${ML_INTERNAL_BASE_PATH}/anomaly_detectors/${jobId}/results/overall_buckets`,
-        method: 'POST',
-        body,
-        version: '1',
+
+      // Max permitted job_id is 64 characters, so we can fit around 30 jobs per request
+      const maxJobsPerRequest = 30;
+
+      return Promise.all(
+        chunk(jobId, maxJobsPerRequest).map((jobIdsChunk) => {
+          return httpService.http<estypes.MlGetOverallBucketsResponse>({
+            path: `${ML_INTERNAL_BASE_PATH}/anomaly_detectors/${jobIdsChunk.join(
+              ','
+            )}/results/overall_buckets`,
+            method: 'POST',
+            body,
+            version: '1',
+          });
+        })
+      ).then((responses) => {
+        // Merge responses
+        return responses.reduce(
```

Comment:
Nit, to avoid the type assertion a few line down.
```suggestion
        return responses.reduce<estypes.MlGetOverallBucketsResponse>(
```

Reply:
updated in 59edb0f4e0754ae94d93625a7fba45c33eaf6f26